### PR TITLE
8277560: Remove WorkerDataArray::_is_serial

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple change of removing effectively dead code.

Test: build

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277560](https://bugs.openjdk.java.net/browse/JDK-8277560): Remove WorkerDataArray::_is_serial

### Reviewers
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**) ⚠️ Review applies to 7138fb8d5b617d15967d2945f95d758ce1a7a710
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6502/head:pull/6502` \
`$ git checkout pull/6502`

Update a local copy of the PR: \
`$ git checkout pull/6502` \
`$ git pull https://git.openjdk.java.net/jdk pull/6502/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6502`

View PR using the GUI difftool: \
`$ git pr show -t 6502`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6502.diff">https://git.openjdk.java.net/jdk/pull/6502.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/workerDataArray.inline.hpp [hunk 1]
```diff
@@ -158,39 +156,31 @@ void WorkerDataArray<T>::set_all(T value) {
...
-  }
+  out->print("%-30s", title());
 
   uint start = 0;
   while (start < _length && get(start) == uninitialized()) {
     start++;
   }
   if (start < _length) {
-    if (_is_serial) {
-      WDAPrinter::summary(out, get(0));
```
Comment (kstefanj) [inline src/hotspot/share/gc/shared/workerDataArray.inline.hpp:173]: From what I can tell these two function will no longer be used after your change:
```
WDAPrinter::summary(outputStream* out, double time);
WDAPrinter::summary(outputStream* out, size_t value);
```
Please remove them.

**Review by kstefanj — APPROVED** [review]
Looks good, but please do the additional removal mentioned below.

File: src/hotspot/share/gc/shared/workerDataArray.inline.hpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/shared/workerDataArray.inline.hpp:173]: Removed.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**